src/main/scala/com/campudus/tableaux/controller/TableauxController.scala

```diff
@@ -1,114 +1,116 @@
-    tableaux.getColumn(tableId, columnId)
-  }
-
-  def getRow(tableId: IdType, rowId: IdType): Future[DomainObject] = {
-    checkArguments(greaterZero(tableId), greaterZero(rowId))
-    verticle.logger.info(s"getRow $tableId $rowId")
-    tableaux.getRow(tableId, rowId)
-  }
-
-  def getCell(tableId: IdType, columnId: IdType, rowId: IdType): Future[DomainObject] = {
-    checkArguments(greaterZero(tableId), greaterZero(columnId), greaterZero(rowId))
-    verticle.logger.info(s"getCell $tableId $columnId $rowId")
-    tableaux.getCell(tableId, columnId, rowId)
-  }
-
-  def deleteTable(tableId: IdType): Future[DomainObject] = {
-    checkArguments(greaterZero(tableId))
-    verticle.logger.info(s"deleteTable $tableId")
-    tableaux.deleteTable(tableId)
-  }
-
-  def deleteColumn(tableId: IdType, columnId: IdType): Future[DomainObject] = {
-    checkArguments(greaterZero(tableId), greaterZero(columnId))
-    verticle.logger.info(s"deleteColumn $tableId $columnId")
-    tableaux.removeColumn(tableId, columnId)
-  }
-
-  def deleteRow(tableId: IdType, rowId: IdType): Future[DomainObject] = {
-    checkArguments(greaterZero(tableId), greaterZero(rowId))
-    verticle.logger.info(s"deleteRow $tableId $rowId")
-    tableaux.deleteRow(tableId, rowId)
-  }
-
-  def fillCell[A](tableId: IdType, columnId: IdType, rowId: IdType, value: A): Future[DomainObject] = {
-    checkArguments(greaterZero(tableId), greaterZero(columnId), greaterZero(rowId), notNull(value, "value"))
-    verticle.logger.info(s"fillCell $tableId $columnId $rowId $value")
-    tableaux.insertValue(tableId, columnId, rowId, value)
-  }
-
-  def resetDB(): Future[DomainObject] = {
-    verticle.logger.info("Reset database")
-    tableaux.resetDB()
-  }
-
-  def changeTableName(tableId: IdType, tableName: String): Future[DomainObject] = {
-    checkArguments(greaterZero(tableId), notNull(tableName, "TableName"))
-    verticle.logger.info(s"changeTableName $tableId $tableName")
-    tableaux.changeTableName(tableId, tableName)
-  }
-
-  def changeColumn(tableId: IdType, columnId: IdType, columnName: Option[String], ordering: Option[Ordering], kind: Option[TableauxDbType]): Future[DomainObject] = {
-    checkArguments(greaterZero(tableId), greaterZero(columnId))
-    verticle.logger.info(s"changeColumn $tableId $columnId $columnName $ordering $kind")
-    tableaux.changeColumn(tableId, columnId, columnName, ordering, kind)
-  }
-
+package com.campudus.tableaux.controller
+
+import com.campudus.tableaux.ArgumentChecker._
+import com.campudus.tableaux.database.Tableaux._
+import com.campudus.tableaux.database._
+import com.campudus.tableaux.database.structure.{TableauxDbType, CreateColumn}
+import com.campudus.tableaux.helper.StandardVerticle
+import org.vertx.scala.platform.Verticle
+
+import scala.concurrent.Future
+
+class TableauxController(val verticle: Verticle, val database: DatabaseConnection) extends DatabaseAccess with StandardVerticle {
```

Comment:
This shouldn't need the `DatabaseConnection` directly. Maybe pass it `Tableaux` instead?